_posts/2021-01-05-git-rebase-merge.md

```diff
@@ -0,0 +1,202 @@
+image:
+  auto: 0050-git
+---
+
+In this article we are going to discuss some of the very important commands from [git](https://git-scm.com/) and [GitHub](https://github.com), and how they make the life of developers easy working individually or working in a team, and when to utilize each of them. We will be comparing Git Rebase and Git Merge, and further, we will explore the potential opportunities to incorporate the basic workflow of Git. We will start from the very basic, and then will keep on leveling up.
+
+If you are a beginner to git or GitHub and are looking to understand some of the basic commands used with git like fork, pull, checkout and commit then you should definitely give this [amazing article](https://reflectoring.io/github-fork-and-pull/) a read.
+
+
+
+- Introduction to Git
+- What is Git Rebase?
+- Working of Git Rebase
+- What is Git Merge?
+- Working of Git Merge
+- Git Rebase vs Git Merge
+- Merging Pros and Cons
+- Choosing the Right Method
+- Common Myths
+- Summary
+
+So bear with me. I hope this will be fun and a learning experience for all those getting their hands dirty with Git 😊
+
+## Introduction to Git
+
+An **Open-Source Distributed Version Control System** is known as Git. It can be broken down into the following major components,
+
+**Control System:** It acts as a content tracker for Git. So, basically, Git can be used to store content – it is usually used to store the code, but the other content can also be stored.
+
+**Version Control System:** Since the developers code in parallel so the version control system helps in maintaining a history of changes by providing the features like branches and merges.
+
+**Distributed Version Control System:** The code in the case of Git is present in two types of repositories – the local, and the central. Managing and synchronizing the local and central repositories comes under the shelter of a Distributed Version Control System.
+
+## What is Git Rebase?
+
+Let's have a look at the concept of Git Rebase. It is the way of migrating or combining a sequence of commits to a new base commit. If one considers it in the context of feature branching workflow, then it is more useful and easily visualized. The figure given below describes the general process of Git Rebasing.
+
+![Git Rebasing](/assets/img/posts/git-rebase-merge/git-rebasing-basic.png)
+
+## Working of Git Rebase
+
+Let's understand the working of Git Rebase by taking a history like A history with a topic branch off another topic branch, for example, we have branched a topic branch (server), and added some server-side functionality to our project, and then made a commit. Now it's time to branch off to make client-side changes by committing a few times. Finally, we go back to the server-side branch and commit a few more changes.
+
+![Git Rebase Example](/assets/img/posts/git-rebase-merge/git-rebase-working1.png)

```

Comment:
Since I have been using these from sources from the internet, it would not be possible to change the master to main.
Let me know if we can go with this although I am a huge supporter of equality, diversity, and inclusion, love this change by GitHub to be inclusive enough to reduce the reference to master and slave.
Do share how we can go with or without these images?

Reply:
Images updated as per your suggestions